cucumber-selenium-tests/src/test/java/com/skillstorm/PageObjects/TaxPage.java

```diff
@@ -0,0 +1,452 @@
+            PIStreetNameField.sendKeys(streetAddress);
+            PICityNameField.sendKeys(city);
+            PIZipcodeField.sendKeys(zipcode);
+            PIStateNameField.sendKeys(state);
+            PIDOBField.sendKeys(DOB);
+            PISSNField.sendKeys(SSN);
+            
+    }
+
+    public void clickPIsubmitButton(String pisave, String next){
+        clickButton(pisave);
+            clickButton(next);
+    }
+
+    public void clickW2StarterButtons(String w2add, String w2edit){
+        clickButton(w2add);
+        clickButton(w2edit);
+    }
+    
+
+    public void fillOutW2Information(String state,String employer,String wages,String federalTax,String stateTax,String SSNtax,
+                                         String medicareTax){
+            w2StateField.sendKeys(state);
+            w2EmployerField.sendKeys(employer);
+            w2WageField.sendKeys(wages);
+            w2FederalIncomeTaxWithheldField.sendKeys(federalTax);
+            w2StateIncomeTaxWithheldField.sendKeys(stateTax);
+            w2SocialSecurityTaxWithheldField.sendKeys(SSNtax);
+            w2medicareTaxWithheldField.sendKeys(medicareTax);
+            
+    }
+
+    public void clickW2submitButton(String w2submit, String next){
+        clickButton(w2submit);
+        clickButton(next);
+    }
+
+    public void fillOutOtherIncomeInformation(String longTermCapitalGains,String shortTermGains,String otherInvestments,
+                                                String netBusinessIncome,String additionalIncome){
+                
+            OILongTermCaptialGainsField.sendKeys(longTermCapitalGains);
+            OIShortTermCaptialGainsField.sendKeys(shortTermGains);
+            OINetBusinessIncomeField.sendKeys(netBusinessIncome);
+            OIOtherInvestmentIncomeField.sendKeys(otherInvestments);
+            OIAdditionalIncomeField.sendKeys(additionalIncome);
+     }
+
+     public void clickOtherIncomeSubmitButton(String oisave, String next){
+        clickButton(oisave);
+        clickButton(next);
+     }
+
+     public void clickDeductionsStarterButtons(String deductAdd, String deductEdit){
+        clickButton(deductAdd);
+        clickButton(deductEdit);
+     }
+
+     public void fillOutDeductions(String deductionType,String deductionAmount ){
+                deductDeductionTypeField.sendKeys(deductionType);
+                deductDeductionAmountField.sendKeys(deductionAmount);
+     }
+
+     public void clickDeductionsSubmitButton(String deductSubmit, String next){
+        clickButton(deductSubmit);
+        clickButton(next);
+     }
+
+     public void clickReviewAndSubmit(String reviewSubmit){
+        clickButton(reviewSubmit);
+     }
+
+    public boolean CheckForEstimationResultsPage(){
+        
+        return driver.findElements(By.xpath("//div[contains(text(),'Congratulations')]")).size() > 0;
+
+        
+    }
+
+
+    @Override
+    public WebElement getWebElement(String name) {
```

Comment:
Medium Impact: this method does not provide functionality to retrieve any WebElement contained within this component, which could cause difficulty in retrieving specific WebElements as needed by other classes. 
Suggestion: Add the missing WebElements.
Note: This method seems to retrieve any given button on the page, which would be great for the getButtons method.

Reply:
This one I am afraid I need more clarification on. This is one of the two I initially used.